admin/lint_json.m

```diff
@@ -0,0 +1,67 @@
+function lint_json(filesin, outputfile)
+% Use mlint, and convert to a json for easy parsing by WNG
+%
+% Input:
+% ------
+% filesin              char array OR cell array of char arrays detailing files to parse
+%                          if filesin is empty will recurse from current working directory
+% outputfile           char array of filename to write output to (will overwrite)
+%                          if outputfile is empty will write to stdout
+
+    if ~exist('filesin', 'var') || (exist('filesin', 'var') && isempty(filesin))
+        % Default to glob all
+        filesin = {['**', filesep, '*.m']};
+    end
+    if ~exist('outputfile', 'var') || (exist('outputfile', 'var') && isempty(outputfile))
+        fh = 1;
+    else % Open file
+        fh = fopen(outputfile,'w');
+        if fh == -1
+            error("MATLAB:FileOpenError", "Failed to open file %s", outputfile);
+        end
+        cleanup = onCleanup(@()(fclose(fh)));
+    end
+
+    files = [];
+    for i = 1:numel(filesin)
+        flist = dir(filesin{i});
+        % Filter doc files
```

Comment:
would it be helpful to pull this "filter" out as a helper function? It's well defined and simply takes an array and returns an array with (currently one) filter applied?

This function is starting to do a number of discrete processing steps which increases its complexity.